eslint.config.mjs

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import prettierConfig from 'eslint-config-prettier';
+import config from '@zendeskgarden/eslint-config';
+import noticePlugin from '@zendeskgarden/eslint-config/plugins/notice.js';
+import reactPlugin from '@zendeskgarden/eslint-config/plugins/react.js';
+import typescriptPlugin from '@zendeskgarden/eslint-config/plugins/typescript.js';
+import typescriptTypeCheckedPlugin from '@zendeskgarden/eslint-config/plugins/typescript-type-checked.js';
```

Comment:
The original config was unable to support type checking, which makes me think there's something misconfigured here. I'm going to dig in locally to double-check.

Reply:
Thanks for looking. Now I know how to check for this in my `react-components` v9 eslint branch 👀